internal/js_parser/js_parser.go

```diff
@@ -73,7 +73,7 @@ type parser struct {
 	shouldFoldNumericConstants bool
 	emittedNamespaceVars       map[js_ast.Ref]bool
 	isExportedInsideNamespace  map[js_ast.Ref]js_ast.Ref
-	knownEnumValues            map[js_ast.Ref]map[string]float64
+	knownEnumValues            map[js_ast.Ref]map[string]js_ast.E
```

Comment:
This is to handle both `EString` and `ENumber`, as with `const enum`s both can be inlined.

Any better approaches for how to handle it better would be great 🙂